runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/net/Url.kt

```diff
@@ -104,6 +105,21 @@ private fun encodePath(
  */
 public data class UserInfo(public val username: String, public val password: String)
 
+/**
+ * Identifies the type of decoding behavior desired when parsing a URL.
+ */
+public enum class UrlDecodingBehavior {
```

Comment:
**comment**: This has odd semantics since component may be from path or query parameters. Only path decoding consults this enum in the parse implementation which sort of reaffirms the oddness of it.